File: fileNaming.js
function fileNaming(names) {
    let map = {};
    for (let i = 0; i < names.length; i++) {
        if (map[names[i]] === undefined) {
            map[names[i]] = 1;
        } else {
            let j = 1;
            while (true) {
                let tmp = names[i] + '(' + j + ')';
                if (map[tmp] == undefined) {
                    map[tmp] = 1;
                    break;
                }

                j++;
            }
        }
    }

    // console.log(Object.keys(map))
    return Object.keys(map);
}

File: fileNaming.py
def fileNaming(names):
    dictionary = {}
    for name in names:
        if name not in dictionary:
            dictionary[name] = 1
        else:
            j = 1
            while True:
                temp = name + "(" + str(j) + ")"
                if temp not in dictionary:
                    dictionary[temp] = 1
                    break
                j += 1
                pass
    print(dictionary)
    return list(dictionary.keys())
